packages/react/src/components/Geo/AmplifyMap/index.tsx

```diff
@@ -0,0 +1,75 @@
+import type { ICredentials } from '@aws-amplify/core';
+import { Amplify, Auth } from 'aws-amplify';
+import maplibregl from 'maplibre-gl';
+import { AmplifyMapLibreRequest } from 'maplibre-gl-js-amplify';
+import React, { useEffect, useState } from 'react';
+import ReactMapGL from 'react-map-gl';
+import type { MapProps, TransformRequestFunction } from 'react-map-gl';
+
+/**
+ * The `AmplifyMap` component uses [react-map-gl](https://visgl.github.io/react-map-gl/) and
+ * [maplibre-gl-js](https://visgl.github.io/react-map-gl/) to provide an interactive map using
+ * [Amplify Geo APIs](https://docs.amplify.aws/lib/geo/getting-started/q/platform/js/) powered by
+ * [Amazon Location Service](https://aws.amazon.com/location/). Since `AmplifyMap` is a wrapper of the
+ * [react-map-gl default Map](https://visgl.github.io/react-map-gl/docs/api-reference/map), it accepts the same
+ * properties.
+ *
+ * @example
+ * // Basic usage of AmplifyMap:
+ * function App() {
+ *   return <AmplifyMap />
+ * }
+ */
+export const AmplifyMap = ({ style, ...props }: MapProps) => {
+  const [transformRequest, setTransformRequest] = useState<
+    TransformRequestFunction | undefined
+  >();
+  const [credentials, setCredentials] = useState<ICredentials | undefined>();
+
+  const amplifyConfig = Amplify.configure() as any;
+  const mapId = amplifyConfig.geo?.amazon_location_service.maps.default;
+  const region = amplifyConfig.geo?.amazon_location_service.region;
+
+  const styleProps = {
+    height: '100vh',
+    position: 'relative',
+    width: '100vw',
+    ...style,
+  } as React.CSSProperties;
+
+  useEffect(() => {
+    const getCredentials = async () => {
+      setCredentials(await Auth.currentUserCredentials());
+    };
+
+    getCredentials();
+  }, []);
+
+  /**
+   * The transformRequest is a callback used by react-map-gl before it makes a request for an external URL. It signs
+   * the request with AWS Sigv4 Auth, provided valid credentials, and is how we integrate react-map-gl with Amplify Geo
+   * and Amazon Location Service. Once the transformRequest is created, we render the map.
+   */
+  useEffect(() => {
+    if (credentials) {
+      const { transformRequest: amplifyTransformRequest } =
+        new AmplifyMapLibreRequest(credentials, region);
+      setTransformRequest(() => amplifyTransformRequest);
+    }
+  }, [credentials]);
+
+  /**
+   * The mapLib property is used by react-map-gl@v7 to override the underlying map library. The default library is
+   * mapbox-gl-js, which uses its own copyrighted license. We override the map library with the BSD-licensed
+   * maplibre-gl-js.
+   */
+  return transformRequest ? (
+    <ReactMapGL
+      mapLib={maplibregl}
+      mapStyle={mapId}
+      style={styleProps}
+      transformRequest={transformRequest}
+      {...props}
```

Comment:
Just a general callout regarding placing spread props as the last prop in a React component. 


Any time a spread `{...props}` is placed as the last attribute on a React component, we override any props that have been listed above it that have matching keys. So in this case if i used an `AmplifyMap` in my app like so:

```
<AmplifyMap transformRequest={() => null} />
```

it would prevent the `transformRequest` value we are specifying on ln 71 from being passed to the underlying `ReactMapGL`. This may be intentional, as I don't have a huge amount of context here, but figured I should note it just in case :)

Reply:
After thinking about this a bit, I think it makes sense for us to omit a user-provided `transformRequest` property since it's the bread and butter of our component and overwriting it defeats the purpose of our library.

We can keep the logic to allow overwriting `mapLib` and `mapStyle` as these are just default values we provide out of the box and a user may want to change them while still taking advantage of Amazon Location Service APIs.